Adds cloud connections target discovery docs (#1616)
OctopusDeploy/docs: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md

To perform discovery of targets from cloud resources, Octopus uses a well-known set of variables configured within your project to provide the credentials for authentication to the cloud provider. Variables can be [scoped](/docs/projects/variables/index.md#scoping-variables) allowing for different credentials to be used when performing discovery as required (for example in different environments).

:::hint

> **Octopus 2022.3** has support for configuring the well-known variables for discovery from when editing your deployment process in the Portal.

Reviewer: ```suggestion
**Octopus 2022.3** and higher supports configuring the well-known variables used for target discovery when editing your deployment process in the Portal.
```

| ----------------------- | -------- | ------------------------------------------------------------------------------------------------------------- |
| `Octopus.Azure.Account` | Y        | An [Azure account](/docs/projects/variables/azure-account-variables.md) to use when discovering cloud targets |

From **Octopus 2022.3**, Azure steps that support target discovery will allow you to configure the variables above from within the step configuration if they have not been configured within your project yet.

> - Open the deployment process and navigate to the Azure step

Reviewer: ```suggestion
1. Open the deployment process and navigate to the Azure step.
```

1. Expand the **Cloud Connections** section.

> 1. Click the **Azure** tile. If target discovery variables have not been configured, this will say **Configure Now**.

Author: Thanks, I've applied this to both sections

> - Open the deployment process and navigate to the Azure step
> - Open the Cloud Connections section
> - Click the Azure tile. If target discovery variables have not been configured this will say "Configure Now".
> - Enter the credentials in the drawer
> - Click Ok. This will save the well-known variables above into your project.

Reviewer: I think a numbered list works here better.

```suggestion
1. Open the deployment process and navigate to the Azure step.
1. Expand the **Cloud Connections** section.
1. Click the Azure tile. If target discovery variables have not been configured, this will say **Configure Now**.
1. Select the Azure Account to use.
1. Click **Ok**. This will save the well-known variables above into your project.
```

Also, just as an aside, does the Azure Account dropdown show accounts from the Infra tab, or a list of Azure Account variables? If it's the latter, it's interesting that we don't support (at least in the screenshot) binding to a variable like you can with the Account section for an Azure script step for example

Author: > I think a numbered list works here better.

Thanks, I'll update :) 

> does the Azure Account dropdown show accounts from the Infra tab or a list of Azure Account variables?

The dropdown only shows accounts from the infra tab. Saving the account sets the well-known variable `Octopus.Azure.Account` that is used for target discovery. We decided not to allow binding to an existing variable from within the drawer at the moment as the user flow is primarily targeted at newer users who want to discover cloud targets, so we wanted to keep it as simple as possible. Existing users already using variables to manage their accounts within steps are likely to already be a bit more advanced in their usage and so if they wanted to bind our well-known variable to a different one the intention is they can use the variables screen for that.

Reviewer: Makes sense. Thank you for the detailed info

> ![](configure-azure-target-discovery.png "width=500")

Reviewer: ```suggestion
![Configuring Cloud Connections for an Azure step](configure-azure-target-discovery.png "width=500")
```

> 

Author: Yep good call, I've added a line to both Azure and Aws sections for this.

| `Octopus.Aws.AssumedRole.SessionDuration` | N        | The maximum duration the session will be available for if assuming a role during discovery. If not set then the default duration from the IAM role will be used.                                                                                             |
| `Octopus.Aws.AssumedRole.ExternalId`      | N        | An external ID to use if assuming a role during discovery. See the AWS documentation for more information on the use of external IDs.                                                                                                                        |

From **Octopus 2022.3**, AWS steps that support target discovery will allow you to configure the variables above from within the step configuration if they have not been configured within your project yet:

> - Open the deployment process and navigate to the AWS step
> - Open the Cloud Connections section
> - Click the Amazon Web Services tile. If target discovery variables have not been configured this will say "Configure Now".
> - Enter the credentials in the drawer
> - Click Ok. This will save the well-known variables above into your project.

Reviewer: ```suggestion
1. Open the deployment process and navigate to the AWS step
1. Expand the **Cloud Connections section.
1. Click the **Amazon Web Services** tile. If target discovery variables have not been configured, this will say **Configure Now**.
1. Choose the credentials to use for target discovery.
1. Click **Ok**. This will save the well-known variables above into your project.
```

> ![](configure-aws-target-discovery.png "width=500")

Reviewer: ```suggestion
![Configuring Cloud Connections for an AWS step](configure-aws-target-discovery.png "width=500")
```
